applications/portfolioDrafts/portfolio/createPortfolioStep.ts

```diff
@@ -20,9 +21,28 @@ export const handler = async (event: APIGatewayProxyEvent): Promise<APIGatewayPr
     );
   }
 
-  const portfolioId = event.pathParameters?.portfolioDraftId;
-
+  const portfolioDraftId = event.pathParameters?.portfolioDraftId;
+  if (!portfolioDraftId) {
+    return new ErrorResponse(
+      { code: ErrorCodes.INVALID_INPUT, message: "PortfolioDraftId must be specified in the URL path" },
+      ErrorStatusCode.BAD_REQUEST
+    );
+  }
+  if (!IsValidJson(event.body)) {
+    return new ErrorResponse(
+      { code: ErrorCodes.INVALID_INPUT, message: "Invalid request body: Invalid JSON" },
+      ErrorStatusCode.BAD_REQUEST
+    );
+  }
   const requestBody = JSON.parse(event.body);
+
+  if (!isPortfolioStep(requestBody)) {
+    return new ErrorResponse(
+      { code: ErrorCodes.INVALID_INPUT, message: "Invalid request body: Missing request attributes" },
+      ErrorStatusCode.BAD_REQUEST
+    );
+  }
+  const now = new Date().toISOString();
   const portfolioStep: PortfolioStep = {
```

Comment:
Will type script let you now do:

```suggestion
  const portfolioStep = requestBody as PortfolioStep;
```

Since it's seen us return if `!requestBody is PortfolioStep`?

Reply:
Don't accept this comment btw. It'll be invalid bc of the next few lines.

Reply:
Not sure what you mean on this comment,